src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RADIUS_DEFAULT = 10;
+    private static final String COLOR_DEFAULT = "WHITE";
+    protected Random random = new Random();
+
+    protected Figure getRandomFigure(int randomFigure) {
+        switch (randomFigure) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomTrapezoid();
+            case 2:
+                return getRandomRectangle();
+            case 3:
+                return getRandomRightRectangle();
+            case 4:
+                return getRandomSquare();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(RADIUS_DEFAULT, COLOR_DEFAULT);
+    }
+
+    private Circle getRandomCircle() {
+        return new Circle(random.nextInt(),
+                new ColorSupplier().getRandomColor());
```

Comment:
```suggestion
        return new Circle(random.nextInt(),
                colorSupplier.getRandomColor());
```
lets make it a class level variable - to avoid creating a new instance each time this method is invoked

Reply:
@[okuzan](https://github.com/okuzan) okuzan [yesterday](https://github.com/mate-academy/jv-oop-advanced/pull/892#discussion_r1093417344)
move this code to methods responsible for it. Also, methods should be private and placed after public methods ?????
Pay attantion, I had those desicion before, but Oleg Kuzan asked change request of it